File: huipyo/src/week05/boj1912/Main.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
let input = fs.readFileSync(filePath).toString().trim().split('\n');
let n = parseInt(input.shift());
var array = input.shift().split(' ').map(Number);
var sum = 0;
var answer = Number.MIN_SAFE_INTEGER;

for (let number of array) {
    sum = Math.max(sum + number, number);
    answer = Math.max(sum, answer);
}
console.log(answer);

File: huipyo/src/week05/boj1912/Main.java
package week05.boj1912;

import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] array = new int[n];
        int sum = 0;
        int answer = Integer.MIN_VALUE;

        Arrays.setAll(array, i -> scanner.nextInt());
        for (int number : array) {
            sum = Math.max(sum + number, number);
            answer = Math.max(sum, answer);
        }
        System.out.println(answer);
        scanner.close();
    }
}